File: self/S2_07052025/ConvertPython/Ex2_Python/Ex2_1.py
import cv2
import matplotlib.pyplot as plt

img = cv2.imread('nature.jpg')
img_rgb = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
img_gray = cv2.cvtColor(img, cv2.COLOR_BGR2GRAY)
_, img_bw = cv2.threshold(img_gray, 127, 255, cv2.THRESH_BINARY)

plt.figure(figsize=(12, 4))
plt.subplot(1, 3, 1), plt.imshow(img_rgb), plt.title("Original Image", fontsize=15), plt.axis('off')
plt.subplot(1, 3, 2), plt.imshow(img_gray, cmap='gray'), plt.title("Grayscale Image", fontsize=15), plt.axis('off')
plt.subplot(1, 3, 3), plt.imshow(img_bw, cmap='gray'), plt.title("Binary Image", fontsize=15), plt.axis('off')
plt.show()
plt.savefig("Image_in_different_type.jpg")

cv2.imwrite('nature100.jpg', img)
cv2.imwrite('nature75.jpg', img, [int(cv2.IMWRITE_JPEG_QUALITY), 75])
cv2.imwrite('nature10.jpg', img, [int(cv2.IMWRITE_JPEG_QUALITY), 10])

File: self/S2_07052025/Ex2_1.m
% Read and Display an image with different quality
clear, clc, close all;
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
pkg load image; %load image package in Octave

Img = imread('nature.jpg'); % color img
Img_grayscale = rgb2gray(Img); %gray scale img
Img_bw = im2double(im2bw(Img_grayscale, 0.5)); %binary

%  display image in different quality
FS = 15; % fontsize of caption
figure, clf;
subplot(1, 3, 1), imshow(Img), title('Original Image', "fontsize", FS); %row, column, vi tri
subplot(1, 3, 2), imshow(Img_grayscale), title('Grayscale Image', "fontsize", FS);
subplot(1, 3, 3), imshow(Img_bw), title('Binary Image', "fontsize", FS);
saveas(gcf, 'Image in different type.jpg');

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%save the origin color image in different quality
imwrite(Img, 'nature100.jpg'); % save as default
imwrite(Img, 'nature75.jpg'); % save as 75% of the origin
imwrite(Img, 'nature10.jpg'); % save as 10% of the origin
